### ISSUE TITLE
Consider cross compiling for 2.10

### ISSUE THREAD

**Issue by neowulf33** [MonsantoCo/cloudformation-template-generator#162]
Hello!

Love this project! Exactly what I was looking for - I wanted to create a SBT plugin for my project which when executing `release` task - would also auto generate the cloudformation task definition templates. This sbt plugin would glean the necessary information from the build settings to generate the template.

However, more than a quarter of my sbt plugins have not been cross compiled for 2.11.

Would it be possible to compile and release this library against scala 2.10?

Thank you for considering my request!

Using version: 

```
"com.monsanto.arch" % "cloud-formation-template-generator_2.11" % "3.6.0"
```

**Comment by neowulf33** [comment]
The incompatibility is shown in the following stacktrace:

```
java.lang.NoSuchMethodError: scala.Predef$.ArrowAssoc(Ljava/lang/Object;)Ljava/lang/Object;
        at com.monsanto.arch.cloudformation.model.resource.ContainerDefinition$$anon$1.write(ECS.scala:151)
        at com.monsanto.arch.cloudformation.model.resource.ContainerDefinition$$anon$1.write(ECS.scala:148)
        at spray.json.PimpedAny.toJson(package.scala:39)
        at spray.json.CollectionFormats$$anon$4$$anonfun$write$4.apply(CollectionFormats.scala:87)
        at spray.json.CollectionFormats$$anon$4$$anonfun$write$4.apply(CollectionFormats.scala:87)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
        at scala.collection.AbstractTraversable.map(Traversable.scala:105)
        at spray.json.CollectionFormats$$anon$4.write(CollectionFormats.scala:87)
        at spray.json.CollectionFormats$$anon$4.write(CollectionFormats.scala:86)
        at spray.json.ProductFormats$class.productElement2Field(ProductFormats.scala:46)
        at com.monsanto.arch.cloudformation.model.resource.AWS$colon$colonECS$colon$colonTaskDefinition$.productElement2Field(ECS.scala:83)
        at spray.json.ProductFormatsInstances$$anon$7.write(ProductFormatsInstances.scala:181)
        at spray.json.ProductFormatsInstances$$anon$7.write(ProductFormatsInstances.scala:176)
        at spray.json.PimpedAny.toJson(package.scala:39)
....
```

**Comment by neowulf33** [comment]
Running into another issue:

```
```

**Comment by bkrodgers** [comment]
My main concern with trying to target 2.10 is that I'm worried it may limit our ability to move forward with current Scala versions.  I generally think it's reasonable to support current and one version prior.  Scala 2.10 is more than 4 years old now, and hasn't had a significant point release in over 2 years (2.10.6 doesn't count, it was just a minor licensing tweak).  It's considered obsolete as far as Scala's maintainers are concerned.  Do you really have that many plugins that aren't even compatible with 2.11, much less 2.12?

That being said, if cross compiling to 2.10 can be done without breaking any current functionality or making any compromises on our ability to move forward with 2.11 and 2.12, I'm open to doing it.  Have you tried cross compiling it yet?  I may not have time in the immediate future to work on it, but if you want to look into the changes needed to add 2.10 support (retaining 2.11 and 2.12), I'd welcome a PR for it.

**Comment by neowulf33** [comment]
Thank you for the feedback. I did try cross compiling (wanted to submit a PR) and I noticed there were some features that were not available in 2.10. I understand your concern - and we have moved on with finding a temp solution. We'll revisit this lib when the time is right for us.
